ci/sal_stack_lwip_ci.py

```diff
@@ -0,0 +1,206 @@
+#!/usr/bin/env python
+
+"""
+##############################################################################
+# sa_stack_lwip_ci.py
+#  CI (noclone) automation script to invoke sal-stack-lwip module testing.
+#
+# This script overcomes the following problems:
+# - sal-stack-lwip is dependent on changes in the sal-stack-lwip module that 
+#   havent been published in the yotta registry. Thus when sal-stack-lwip
+#   builds with the yotta published version the build would break. This
+#   is overcome by git cloning sal-stack-lwip (to get HEAD with changes)
+#   and using yt link in sal-stack-lwip to use that version of sal-stack-lwip.
+#
+# Example usage: 
+#     $ sal_stacl_lwip_ci.py 
+#
+# Author: simon.hughes@arm.com
+###############################################################################
+"""
+
+import subprocess
+import argparse
+import sys
+import os      
+import time
+import datetime
+
+# error codes are positive numbers because sys.exit() returns a +ve number to the env
+MBED_SUCCESS = 0
+MBED_FAILURE = 1
+MBED_ERROR_MAX = 2
+
+#g_debug = True
+g_debug = False
+
+# a debug trace function
+def dbg(str):
+    if g_debug:
+        print str
+    return
+
+
+##############################################################################
+# sal_test
+##############################################################################
+class sal_test:
```

Comment:
I'm not clear on the reason for this class.  Why do we need the facility to link/unlink modules?